sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -251,3 +252,11 @@ def __init__(self):
                         check_no_fit_attributes_set_in_init,
                         'estimator_name',
                         NonConformantEstimator)
+
+
+def test_check_estimator_pairwise():
+    # check that check_estimator() works on estimator with _pairwise
+    # attribute set
+    est = SVC(kernel='precomputed')
+    check_estimator(est)
```

Comment:
Can we do this for an estimator based on a metric as well as a kernel? It's very possible that doing so will break.

Reply:
Got it. Can you give a quick example of an estimator based on a metric as well as a kernel?

Reply:
KNeighborsRegressor or AgglomerativeClustering